tracer/src/Datadog.Trace/Debugger/Symbols/SymbolExtractor.cs

```diff
@@ -0,0 +1,718 @@
+
+                        methodScope.LanguageSpecifics = ls;
+                    }
+                }
+
+                classMethods[index] = methodScope;
+                index++;
+            }
+        }
+
+        private SourcePdbInfo GetMethodSourcePdbInfo(Model.Scope methodScope)
+        {
+            var startLine = int.MaxValue;
+            var endLine = 0;
+            int? startColumn = null;
+            int? endColumn = null;
+            string? sourceFile = null;
+            for (int i = 0; i < methodScope.Scopes.Count; i++)
+            {
+                var scope = methodScope.Scopes[i];
+                if (startLine > scope.StartLine && scope.StartLine > 0)
+                {
+                    startLine = scope.StartLine;
+                }
+
+                if (endLine < scope.EndLine)
+                {
+                    endLine = scope.EndLine;
+                }
+
+                sourceFile ??= scope.SourceFile;
+
+                if (!scope.LanguageSpecifics.HasValue)
+                {
+                    continue;
+                }
+
+                if (startColumn == null || startColumn > scope.LanguageSpecifics.Value.StartColumn)
+                {
+                    startColumn = scope.LanguageSpecifics.Value.StartColumn;
+                }
+
+                if (endColumn == null || endColumn < scope.LanguageSpecifics.Value.EndColumn)
+                {
+                    endColumn = scope.LanguageSpecifics.Value.EndColumn;
+                }
+            }
+
+            if (startLine == int.MaxValue)
+            {
+                startLine = UnknownMethodStartLine;
+            }
+
+            if (endLine == 0)
+            {
+                endLine = UnknownMethodEndLine;
+            }
+
+            return new SourcePdbInfo { StartLine = startLine, EndLine = endLine, Path = sourceFile, StartColumn = startColumn, EndColumn = endColumn };
+        }
+
+        protected virtual Model.Scope CreateMethodScope(TypeDefinition type, MethodDefinition method)
+        {
+            // arguments
+            var argsSymbol = GetArgsSymbol(method);
+
+            // closures
+            var closureScopes = GetClosureScopes(type, method);
+            var methodAttributes = method.Attributes & StaticFinalVirtual;
+            var methodLanguageSpecifics = new LanguageSpecifics
+            {
+                ReturnType = method.DecodeSignature(new TypeProvider(false), 0).ReturnType,
+                AccessModifiers = (method.Attributes & MethodAttributes.MemberAccessMask) > 0 ? new[] { _methodAccess[Convert.ToUInt16(method.Attributes & MethodAttributes.MemberAccessMask)] } : null,
+                Annotations = methodAttributes > 0 ? new[] { _methodAttributes[Convert.ToUInt16(methodAttributes)] } : null
+            };
+
+            var methodScope = new Model.Scope
+            {
+                ScopeType = SymbolType.Method,
+                Name = MetadataReader.GetString(method.Name),
```

Comment:
To avoid having scopes without a name, what do you think about checking what `MetadataReader.GetString` returns (`string.IsNullOrEmpty`) and only then include this scope in the batch of symbols?
My concern is that we will upload scopes without names which will complicate things. If we, instead of uploading them, log them and skip them it'll make our lives easier in diagnostics these issues later. I don't even know what would happen in the UI when a scope holds `null` or an empty string in it's name. 🤔

Reply:
It's a good point. Do you think that there is a case when the user will try to put a probe with class name and line number? If the answer is yes, maybe we should upload methods without names?
(Anyway, taking in account that we skip irregular method, I must say that as far as the `method.Name` handle isn't `Nill`, we should get a non empty name.)

Reply:
I don't get it. You say that the line selection is based on symbol information?
I would argue that if we can't find the method name then how _method probes_ will work? 🤔 I don't even know how the backend will handle it. Undefined behavior?

Reply:
I think you're right, I'll skip those methods with no name (though as I wrote, I doubt we'll see those cases).

Reply:
Fixed in d76bac0e9bd3cfbda3e480deff9bbd0e453a32ec